pdh.go

```diff
@@ -349,10 +389,12 @@ func PdhGetFormattedCounterValueLong(hCounter PDH_HCOUNTER, lpdwType *uint32, pV
 //		if ret == win.ERROR_SUCCESS {
 //			ret = win.PdhGetFormattedCounterArrayDouble(counterHandle, &bufSize, &bufCount, &emptyBuf[0]) // uses null ptr here according to MSDN.
 //			if ret == win.PDH_MORE_DATA {
-//				filledBuf := make([]win.PDH_FMT_COUNTERVALUE_ITEM_DOUBLE, bufCount*size)
+//				// Make sure we're allocating the requested amount of space
+//				filledBuf := make([]byte, bufSize)
 //				ret = win.PdhGetFormattedCounterArrayDouble(counterHandle, &bufSize, &bufCount, &filledBuf[0])
```

Comment:
`PdhGetFormattedCounterArray` returns the number of elements to expect (`bufCount`), and the size of the buffer to provide (`bufSize`). Providing both *would* be redundant, but `bufSize` is actually greater than `bufCount * sizeof(PDH_FMT_COUNTERVALUE_ITEM)` because they need some extra space at the end of the buffer for other things (like strings).

The old example is just completely bonkers, it recommended creating a slice of `PDH_FMT_COUNTERVALUE_ITEM` with the size of `bufCount * sizeof(PDH_FMT_COUNTERVALUE_ITEM)`. This will make a really big array (`sizeof(PDH_FMT_COUNTERVALUE_ITEM)` ^ 2 * `bufCount` bytes), but it doesn't guarantee the array will actually be big enough.